src/bmtrap.c

```diff
@@ -254,38 +254,36 @@ void LoadTrapData(const struct TrapData * data)
         while (data->type) {
             switch (data->type) {
                 case TRAP_BALLISTA:
-                    AddBallista(data->xPos, data->yPos, data->unk_03);
+                    AddBallista(data->xPos, data->yPos, data->subtype);
                     break;
 
                 case TRAP_FIRETILE:
-                    AddFireTile(data->xPos, data->yPos, data->unk_04, data->unk_05);
+                    AddFireTile(data->xPos, data->yPos, data->turn_counter, data->turn);
                     break;
 
                 case TRAP_GAS:
-                    AddGasTrap(data->xPos, data->yPos, data->unk_03, data->unk_04, data->unk_05);
+                    AddGasTrap(data->xPos, data->yPos, data->subtype, data->turn_counter, data->turn);
                     break;
 
                 case TRAP_8:
                     AddTrap8(data->xPos, data->yPos);
                     break;
 
                 case TRAP_9:
-                    AddTrap9(data->xPos, data->yPos, data->unk_03);
+                    AddTrap9(data->xPos, data->yPos, data->subtype);
                     break;
 
                 case TRAP_MINE:
                     AddTrap(data->xPos, data->yPos, data->type, 0);
                     break;
 
                 case TRAP_LIGHTARROW:
-                    AddArrowTrap(data->xPos, data->unk_04, data->unk_05);
-                    // break; // BUG?
+                    AddArrowTrap(data->xPos, data->turn_counter, data->turn);
 
```

Comment:
Could you update this to use `#if BUGFIX` here? It seems that this was an omission of the FE8 devs when adding a new switch case to this function.

This was encountered as a bug in the Discord: https://discord.com/channels/144670830150811649/470029781795078175/1219122326122135643

Reply:
updated